ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java

```diff
@@ -280,6 +283,66 @@ public void cleanup() throws InterruptedException, IOException, StoreException {
   }
 
   /**
+   * Tests blob store use of {@link ClusterManagerWriteStatusDelegate}
+   * @throws StoreException
+   */
+  @Test
+  public void testClusterManagerWriteStatusDelegateUse() throws StoreException, IOException, InterruptedException {
+    //Setup threshold test properties, replicaId, mock write status delegate
+    properties.setProperty(StoreConfig.storeReadOnlyEnableSizeThresholdPercentageName, "65");
+    properties.setProperty(StoreConfig.storeReadWriteEnableSizeThresholdPercentageDeltaName, "5");
+    StoreConfig config = new StoreConfig(new VerifiableProperties(properties));
+    StoreTestUtils.MockReplicaId replicaId = getMockReplicaId(tempDirStr);
+    ClusterManagerWriteStatusDelegate clusterManagerWriteStatusDelegate = mock(ClusterManagerWriteStatusDelegate.class);
+    when(clusterManagerWriteStatusDelegate.unseal(anyObject())).thenReturn(true);
+    when(clusterManagerWriteStatusDelegate.seal(anyObject())).thenReturn(true);
+
+    //Restart store with new threshold properties, mock delegate
+    store.shutdown();
+    store = createBlobStore(replicaId, config, clusterManagerWriteStatusDelegate);
+    store.start();
+
+    //Check that after start, because StoreDescriptor file already exists, delegate is not called
+    verifyZeroInteractions(clusterManagerWriteStatusDelegate);
+
+    //Verify that after putting in enough data, the store goes to read only
+    List<MockId> addedIds = put(5, 900, Utils.Infinite_Time);
+    verify(clusterManagerWriteStatusDelegate, times(1)).seal(replicaId);
+
+    //Assumes ClusterParticipant sets replicaId status to true
+    replicaId.setSealedState(true);
+
+    //Remaining tests only relevant for segmented logs
+    if (isLogSegmented) {
+      //Delete added data
+      for (MockId addedId : addedIds) {
+        delete(addedId);
+      }
+
+      //Need to restart blob otherwise compaction will ignore segments in journal (which are all segments right now).
+      //By restarting, only last segment will be in journal
+      store.shutdown();
+      store = createBlobStore(replicaId, config, clusterManagerWriteStatusDelegate);
+      store.start();
+      verifyNoMoreInteractions(clusterManagerWriteStatusDelegate);
+
+      //Advance time by 8 days, call compaction to compact segments with deleted data, then verify
+      //that the store is now read-write
+      time.sleep(TimeUnit.DAYS.toMillis(8));
+      store.compact(store.getCompactionDetails(new CompactAllPolicy(config, time)));
+      verify(clusterManagerWriteStatusDelegate, times(1)).unseal(replicaId);
+
+      //Test when StoreDescriptor is deleted and replicaId is erroneously true that it updates the status upon startup
```

Comment:
do you think its also possible to test the other scenario we discussed (changing of config)?

Reply:
added